Cython/Compiler/ExprNodes.py

```diff
@@ -13589,7 +13588,46 @@ def generate_operation_code(self, code, result_code,
                 operand1.type.unary_op('eq'),
                 operand1.result(),
                 operand2.result()))
-
+        elif operand1.type.is_struct_or_union or operand2.type.is_struct_or_union:
+            assert op in ('==', '!=')
+
+            if not operand1.type.is_struct_or_union or not operand2.type.is_struct_or_union:
+                error(self.pos, "cannot compare struct to non-struct")
+
+            if (operand1.type.is_struct_or_union and operand1.type.kind == 'union') or (operand2.type.is_struct_or_union and operand2.type.kind == 'union'):
+                error(self.pos, "cannot compare unions")
+
+            if not operand1.type.assignable_from(operand2.type):
+                error(self.pos, "cannot compare structs of different types")
+
+            self.operand1_temp = code.funcstate.allocate_temp(operand1.type, manage_ref=False)
+            self.operand2_temp = code.funcstate.allocate_temp(operand1.type, manage_ref=False)
```

Comment:
I think these can just be local variables. Same in the elif section below.

Reply:
Done, this was left over from copied code that used these in a `clear_temps()` function.